src/CitizenInitiative/CitizenInitiativeControllerSubscriber.php

```diff
@@ -0,0 +1,67 @@
+<?php
+namespace AppBundle\CitizenInitiative;
+
+use AppBundle\Controller\EnMarche\CitizenInitiativeController;
+use AppBundle\Controller\EnMarche\CitizenInitiativeManagerContoller;
+use AppBundle\Entity\Adherent;
+use Symfony\Bundle\FrameworkBundle\Routing\Router;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+class CitizenInitiativeControllerSubscriber implements EventSubscriberInterface
+{
+    protected $router;
+    protected $tokenStorage;
+
+    public function __construct(Router $router, TokenStorageInterface $tokenStorage)
+    {
+        $this->router = $router;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    public function onKernelController(FilterControllerEvent $event)
+    {
+        $controller = $event->getController();
+
+        /*
+         * $controller passed can be either a class or a Closure.
+         * This is not usual in Symfony but it may happen.
+         * If it is a class, it comes in array format
+         */
+        if (!is_array($controller)) {
+            return;
+        }
+
+        if ($controller[0] instanceof CitizenInitiativeController || $controller[0] instanceof CitizenInitiativeManagerContoller) {
+            $event->getRequest()->attributes->set('is_ci_controller', true);
+        }
+    }
+
+    public function onKernelResponse(FilterResponseEvent $event)
+    {
+        if ($event->getRequest()->attributes->has('is_ci_controller') && $event->getRequest()->attributes->get('is_ci_controller')) {
+            $connectedUser = $this->tokenStorage->getToken()->getUser();
+            if ($connectedUser && $connectedUser instanceof Adherent) {
+                // check if departement of connected user is authorized
+                // authorized departements : Somme: 80, Paris 18: 75018, Charente: 16, Tarn: 81
+                $codes = ['16', '80', '81'];
+                if ($connectedUser->getCountry() == 'FR' && $connectedUser->getPostalCode()
```

Comment:
what about Yoda conditions ?
`if ('FR' == $connectedUser->getCountry())`


and should we compare variables strictly ?
`if ('FR' === $connectedUser->getCountry())`
and
`in_array(substr($connectedUser->getPostalCode(), 0, 2), $codes, true)`